paper-linear-progress.html

```diff
@@ -0,0 +1,916 @@
+         * be buffered.
+         */
+        buffering: {
+          type: Boolean,
+          observer: '_bufferingChanged'
+        },
+
+        /**
+         * When set, the linear progress element will be hidden from view. Note that this is
+         * _different_ from `hidden`, as it does not set `display: none;` but rather just changes
+         * the element's opacity.
+         */
+        closed: {
+          type: Boolean,
+          observer: '_closedChanged'
+        },
+
+        /**
+         * Animation configuration for NeonAnimationRunnerBehavior.
+         */
+        animationConfig: {
+          value: function() {
+            // NOTE(traviskaufman): This is initialized within the attached() method, once we know
+            // the height of the progress indicator.
+            return {};
+          }
+        },
+
+        __needsBufferingAnimationOnAttached: {
+          // HACK: We want to have buffering play an animation when it's set, but we can't
+          // animate unless we first have the computed height of the element, which we can only
+          // retrieve once the element's attached. Therefore, we set up a queue of animations that
+          // need to be run, and if we encounter any animations. Note that this can be fixed once
+          // https://github.com/Polymer/polymer/issues/2399 is resolved. Then, we can do all
+          // animations within CSS.
+          type: Boolean,
+          value: false
+        }
+      },
+
+      hostAttributes: {
+        role: 'progressbar'
+      },
+
+      ready: function() {
+        this.toggleClass('isIE', /MSIE/.test(navigator.userAgent));
+      },
+
+      attached: function() {
+        this._applyBufferingDotsBackground();
+        this._applyAnimationConfig();
+        if (this.__needsBufferingAnimationOnAttached) {
+          this._playCorrectBufferingAnimation();
+        }
+      },
+
+      _valueChanged: function(newValue) {
+        this._scaleToPercent(newValue, this.$.progressPrimary);
+        if (!this.indeterminate) {
+          this.setAttribute('aria-valuenow', newValue);
+        }
+        this.fire('paper-linear-progress-value-change', {value: newValue});
+      },
+
+      _bufferValueChanged: function(newBufferValue) {
+        this._scaleToPercent(newBufferValue, this.$.buffer);
+        this.fire('paper-linear-progress-buffer-value-change', {bufferValue: newBufferValue});
+      },
+
+      _indeterminateChanged: function(isIndeterminate) {
+        this._forceLayoutIfHasClass('reversed');
+        this.toggleClass('indeterminate', isIndeterminate);
+        if (isIndeterminate) {
+          this.removeAttribute('aria-valuemin');
+          this.removeAttribute('aria-valuemax');
+          this.removeAttribute('aria-valuenow');
+        } else {
+          this.setAttribute('aria-valuemin', 0);
+          this.setAttribute('aria-valuemax', 100);
```

Comment:
I think trying to make paper-slider use paper-linear-progress is going to lead to a lot of headaches down the line (for example, when implementing [range sliders](https://github.com/PolymerElements/paper-slider/issues/69), which is something our designers do want eventually). My suggestion would be to keep them separate, and in the short term have `paper-slider` continue to use `paper-progress`. In the long term, we may be able to refactor some of the functionality from `paper-linear-progress` into a behavior, and use that within both elements.